File: client/src/game/conf/IntervalConfig.ts
class IntervalConfig {
    private static _dic:Dictionary<IntervalConfig> = new Dictionary<IntervalConfig>();
    private static _arr:Array<IntervalConfig> = new Array<IntervalConfig>();

	 public Id :number ;
     public IntervalId :number ;
     public OddsId :number ;
     public McId :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<IntervalConfig>{
        return  IntervalConfig._dic;
    }
    public static get Arr():Array<IntervalConfig>{
        return IntervalConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:IntervalConfig = <IntervalConfig>json[i];
				IntervalConfig._arr.push(config);
				IntervalConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("IntervalConfig loader fail!");
        }
    }
}

File: server/src/game/conf/IntervalConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type IntervalConfigElem struct {
	Id int
	IntervalId int
	OddsId int
	McId int
	
}

type IntervalConfig0 struct {
	Dic map[int]*IntervalConfigElem
	Arr []*IntervalConfigElem
}

var (
	IntervalConfig IntervalConfig0
)

func loadIntervalConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	IntervalConfig.Dic = make(map[int]*IntervalConfigElem)
	IntervalConfig.Arr = make([]*IntervalConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &IntervalConfig.Arr)
	for _,conf := range IntervalConfig.Arr{
		IntervalConfig.Dic[conf.Id] = conf
	}
	
}
